Adding multi cell migration page (#4688)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/migration.md

--- 
title: "Multi-cell migration checklist"
id: migration 
description: "Prepare for account migration to AWS cell based architecture." 
pagination_next: null
pagination_prev: null
---

> dbt Labs is in the process of migrating our U.S. based multi-tenant accounts to [AWS cell-based architecture](https://docs.aws.amazon.com/wellarchitected/latest/reducing-scope-of-impact-with-cell-based-architecture/what-is-a-cell-based-architecture.html), a critical component of the [AWS well-architected framework](https://aws.amazon.com/architecture/well-architected/?wa-lens-whitepapers.sort-by=item.additionalFields.sortDate&wa-lens-whitepapers.sort-order=desc&wa-guidance-whitepapers.sort-by=item.additionalFields.sortDate&wa-guidance-whitepapers.sort-order=desc). The benefits of the cell-based architecture will improve the performance, reliability, and security of your dbt Cloud environment, but there is some preparation required to ensure a successful migration.

Reviewer: i used to write "U.S." as well but i was recently retrained to use "US" instead, as the more modern way of spelling it 🤷‍♀️ 

```suggestion
dbt Labs is in the process of migrating our US based multi-tenant accounts to the [AWS cell-based architecture](https://docs.aws.amazon.com/wellarchitected/latest/reducing-scope-of-impact-with-cell-based-architecture/what-is-a-cell-based-architecture.html), a critical component of the [AWS well-architected framework](https://aws.amazon.com/architecture/well-architected/?wa-lens-whitepapers.sort-by=item.additionalFields.sortDate&wa-lens-whitepapers.sort-order=desc&wa-guidance-whitepapers.sort-by=item.additionalFields.sortDate&wa-guidance-whitepapers.sort-order=desc). The benefits of the cell-based architecture will improve the performance, reliability, and security of your dbt Cloud environment, but there is some preparation required to ensure a successful migration.
```

> dbt Labs is in the process of migrating our U.S. based multi-tenant accounts to [AWS cell-based architecture](https://docs.aws.amazon.com/wellarchitected/latest/reducing-scope-of-impact-with-cell-based-architecture/what-is-a-cell-based-architecture.html), a critical component of the [AWS well-architected framework](https://aws.amazon.com/architecture/well-architected/?wa-lens-whitepapers.sort-by=item.additionalFields.sortDate&wa-lens-whitepapers.sort-order=desc&wa-guidance-whitepapers.sort-by=item.additionalFields.sortDate&wa-guidance-whitepapers.sort-order=desc). The benefits of the cell-based architecture will improve the performance, reliability, and security of your dbt Cloud environment, but there is some preparation required to ensure a successful migration.

Reviewer 2: ```suggestion
dbt Labs is in the process of migrating dbt Cloud to a new **cell-based architecture**. This architecture will be the foundation of dbt Cloud for years to come, and will bring improved **scalability**, **reliability**, and **security** to all customers and users of dbt Cloud.

There is some preparation required to ensure a successful migration.

Migrations are being scheduled on a per-account basis. **If you have not received any communication (either via a banner, or via an email) about a migration date, you do not need to take any action at this time.** Our team will share a specific migration date with you, with appropriate advance notice, before we complete any migration steps in the dbt Cloud backend.
```

Once the [blog post](https://www.notion.so/dbtlabs/Multi-Cell-Enablement-The-Future-of-dbt-Cloud-Introducing-Cell-Based-Architecture-91979d4b369947e78cb924d539d25d01?pvs=4) is live, I'll come back here and make a PR to (a) link it and (b) add a graphic in here explaining what multi-cell is

This document outlines the steps that you must take to prevent service disruptions before your environment is migrated over to the cell-based architecture. This will impact areas such as login, IP restrictions, and API access. 

> ### What’s changing? Pre-migration checklist.

Reviewer: suggs: 

- i generally defer to Merriam Webster as the de facto dictionary for American English https://www.merriam-webster.com/dictionary/premigration#:~:text=%3A%20existing%20or%20occurring%20before%20a,way%20to%20explaining%20group%20success%20%E2%80%A6 
- looks like 2 sentences vs a section title
- H2 vs H3. i've been taught to not skip headings (unless unnecessary to avoid)

```suggestion
## Premigration checklist
```

Author: Good catch on the H3, thank you. The definition linked has "pre-migration" as alternative usage and since that's how it's frequently used by tech articles (including [Microsoft](https://techcommunity.microsoft.com/t5/azure-database-for-postgresql/introducing-pre-migration-validations-for-single-server-to/ba-p/3963366)), I think we should keep the hyphen

> Prior to your migration date, your account admin will need to make some changes to your dbt Cloud account.

Reviewer: sugg: might be good to be explicit as to which account admin we're referring to here, especially since we have requirements for a "Microsoft account admin" on other docs pages, for example

```suggestion
Prior to your migration date, your dbt Cloud account admin will need to make some changes to your account.
```

> If your account has been scheduled for migration, upon login, you will see a banner indicating your migration date. If you do not see a banner, you do not need to take any action.

Reviewer: sugg: slightly more concise

```suggestion
If your account is scheduled for migration, you will see a banner indicating your migration date when you log in. If you don't see a banner, you don't need to take any action.
```

> 1. **IP Addresses** &mdash; dbt Cloud has new IPs that will be used to access your warehouse after the migration. Make sure to allow inbound traffic from these IPs in your firewall, and include it in any database grants. All six of the IPs below should be added to allowlists.

Reviewer: ```suggestion
1. **IP addresses** &mdash; dbt Cloud will be using new IPs to access your warehouse after the migration. Make sure to allow inbound traffic from these IPs in your firewall and include it in any database grants. All six of the IPs below should be added to allowlists.
```

    * Old IPs: `52.45.144.63`, `54.81.134.249`, `52.22.161.231`
    * New IPs: `52.3.77.232`, `3.214.191.130`, `34.233.79.135`

> 2. **APIs and integrations** &mdash; Each dbt Cloud account will be allocated a static Access URL like: `aa000.us1.dbt.com`. You should begin migrating your API access and partner integrations to use the new static subdomain as soon as possible. You can find your Access URL on:

Reviewer: sugg: not a proper noun, lower case a

```suggestion
2. **APIs and integrations** &mdash; Each dbt Cloud account will be allocated a static access URL like: `aa000.us1.dbt.com`. You should begin migrating your API access and partner integrations to use the new static subdomain as soon as possible. You can find your access URL on:
```

    * Any page where you generate or manage API tokens.
    * The **Account Settings** > **Account page**.
        
    :::important Multiple account access

>     Each account for which you have access will have a different, dedicated [Access URL](https://next.docs.getdbt.com/docs/cloud/about-cloud/access-regions-ip-addresses#accessing-your-account)!

Reviewer: it read like a warning initially to me so sugg being explicit about it for easier translation (vs using an exclamation point)

```suggestion
    Be careful, each account that you have access to will have a different, dedicated [access URL](https://next.docs.getdbt.com/docs/cloud/about-cloud/access-regions-ip-addresses#accessing-your-account).
```

    :::

> 3. **IDE sessions** &mdash; Any uncommitted changes in the IDE may be lost during the migration process. We _strongly_ encourage you to commit all changes in the IDE before your scheduled migration time.

Reviewer: ```suggestion
3. **IDE sessions** &mdash; Any uncommitted changes in the IDE might be lost during the migration process. dbt Labs _strongly_ encourages you to commit all changes in the IDE before your scheduled migration time.
```

> 4. **User invitations** &mdash; Any pending user invitations will be invalidated during the migration. You can re-send the invitations once the migration is complete.

Reviewer: per https://www.merriam-webster.com/dictionary/resend

```suggestion
4. **User invitations** &mdash; Any pending user invitations will be invalidated during the migration. You can resend the invitations once the migration is complete.
```

> 5. **Git Integrations** &mdash; Integrations with Github, Gitlab, and Azure DevOps will need to be manually updated. We are not migrating any accounts using these integrations at this time. If you are using one of these integrations and your account is scheduled for migration, please contact support and we will delay your migration.

Reviewer: ```suggestion
5. **Git integrations** &mdash; Integrations with GitHub, GitLab, and Azure DevOps will need to be manually updated. dbt Labs will not be migrating any accounts using these integrations at this time. If you're using one of these integrations and your account is scheduled for migration, please contact support and we will delay your migration.
```

> 6. **SSO Integrations** &mdash; Integrations with SSO IdPs will need to be manually updated. We are not migrating any accounts using SSO at this time; if you are using one of these integrations and your account is scheduled for migration, please contact support, and we will delay your migration.

Reviewer: ```suggestion
6. **SSO integrations** &mdash; Integrations with SSO identity providers (IdPs) will need to be manually updated. dbt Labs will not be migrating any accounts using SSO at this time. If you're using one of these integrations and your account is scheduled for migration, please contact support and we will delay your migration.
```

> ### Post-migration

Reviewer: for parallel construction with the other section, maybe `Postmigration checklist` could work here 🤷‍♀️  

```suggestion
## Post-migration
```

Author: Ssame feedback as above - keeping the hyphen, but good catch on the H3

> After migration, if you completed all of the checklist items above, your dbt Cloud resources and jobs will continue to work as they did before. 

Reviewer: ```suggestion
After migration, if you completed all the [Pre-migration checklist](#pre-migration-checklist) items, your dbt Cloud resources and jobs will continue to work as they did before. 
```

> You have the option to log into dbt Cloud at a different URL:

Reviewer: ```suggestion
You have the option to log in to dbt Cloud using a different URL:
```

Author: Acceptable usage is "log in to" going to leave this as is

 * If you were previously logging in at `cloud.getdbt.com`, you should instead plan to login at `us1.dbt.com`. The original URL will still work, but you’ll have to click through to be redirected upon login.

>  * You may also log in directly with your account’s unique [Access URL](https://next.docs.getdbt.com/docs/cloud/about-cloud/access-regions-ip-addresses#accessing-your-account).

Reviewer: ```suggestion
 * You may also log in directly with your account’s unique [access URL](https://next.docs.getdbt.com/docs/cloud/about-cloud/access-regions-ip-addresses#accessing-your-account).
```

--- 
title: "Multi-cell migration checklist"
id: migration 

> description: "Prepare for account migration to AWS cell based architecture." 

Reviewer: ```suggestion
description: "Prepare for account migration to AWS cell-based architecture." 
```

> description: "Prepare for account migration to AWS cell based architecture." 

Author: ```suggestion
description: "Prepare for account migration to AWS cell-based architecture." 
```

pagination_next: null
pagination_prev: null
---

> dbt Labs is in the process of migrating dbt Cloud to a new **cell-based architecture**. This architecture will be the foundation of dbt Cloud for years to come, and will bring improved **scalability**, **reliability**, and **security** to all customers and users of dbt Cloud.

Reviewer: per style guide, for emphasis we can use italics, callouts, banners, etc vs bold (which are for titles, headings, UI labels, etc)

```suggestion
dbt Labs is in the process of migrating dbt Cloud to a new _cell-based architecture_. This architecture will be the foundation of dbt Cloud for years to come, and will bring improved scalability, reliability, and security to all customers and users of dbt Cloud.
```

There is some preparation required to ensure a successful migration.

> Migrations are being scheduled on a per-account basis. **If you have not received any communication (either via a banner, or via an email) about a migration date, you do not need to take any action at this time.** Our team will share a specific migration date with you, with appropriate advance notice, before we complete any migration steps in the dbt Cloud backend.

Author: ```suggestion
Migrations are being scheduled on a per-account basis. _If you have not received any communication (either via a banner or email notification) about a migration date, you do not need to take any action at this time._ dbt Labs will share migration date information with you, with appropriate advance notice, before we complete any migration steps in the dbt Cloud backend.
```

> Migrations are being scheduled on a per-account basis. _If you have not received any communication (either via a banner or email notification) about a migration date, you do not need to take any action at this time._ dbt Labs will share migration date information with you, with appropriate advance notice, before we complete any migration steps in the dbt Cloud backend.

Reviewer: ```suggestion
Migrations are being scheduled on a per-account basis. _If you haven't received any communication (either with a banner or by email) about a migration date, you don't need to take any action at this time._ dbt Labs will share migration date information with you, with appropriate advance notice, before we complete any migration steps in the dbt Cloud backend.
```

This document outlines the steps that you must take to prevent service disruptions before your environment is migrated over to the cell-based architecture. This will impact areas such as login, IP restrictions, and API access. 

> ## Premigration checklist

Author: ```suggestion
## Pre-migration checklist
```

4. **User invitations** &mdash; Any pending user invitations will be invalidated during the migration. You can resend the invitations once the migration is complete.
5. **Git integrations** &mdash; Integrations with GitHub, GitLab, and Azure DevOps will need to be manually updated. dbt Labs will not be migrating any accounts using these integrations at this time. If you're using one of these integrations and your account is scheduled for migration, please contact support and we will delay your migration.
6. **SSO integrations** &mdash; Integrations with SSO identity providers (IdPs) will need to be manually updated. dbt Labs will not be migrating any accounts using SSO at this time. If you're using one of these integrations and your account is scheduled for migration, please contact support and we will delay your migration.

## Post-migration

After migration, if you completed all the [Pre-migration checklist](#pre-migration-checklist) items, your dbt Cloud resources and jobs will continue to work as they did before. 

> You have the option to log into dbt Cloud at a different URL:

Author: ```suggestion
You have the option to log in to dbt Cloud at a different URL:
```
